### DISCUSSION TITLE
feature request ideas

### DISCUSSION THREAD

**Discussion by despiegk** [docusealco/docuseal#3] [Ideas]
first allow me to thank you for the amazing app you have created

some ideas of things we would like to see and potentially even help realize

- use a single sign on solution like threefold connect (see www.threefold.io) on top of signature field
- every user can see the docs they created/signed so its easier tot track what happens
- API to allow to read the filled in properties like checkbox, text field, ... so another can know what was filled in

must have before we can use it

- an API so its easy to integrate in other stacks (not for the creation of the docs, thats ok, but invitations, follow up...)
- verification of email addr

we would love to collaborate and host it on threefold as well

regards

Kristof

**Comment by james-apex27** [comment]
I'd like to second the API request.

For an account allow them to create an API Key

With the API key allow us to call various end-points:-

1. An endpoint to get a list of templates
2. An endpoint to get information about a template (JSON object with signer/field information)
3. An endpoint to to request a signature, passing over the signers information and prefilling fields (such as name, dates, etc) for each signer.
4. An endpoint to get a list of submissions, with JSON object indicating who has signed so far.

It would also be very useful to be able to register a webhook for when a document is signed by someone and when a document is completed signed
